packages/extension/src/ui/features/shield/ShieldBaseFinishScreen.tsx

```diff
@@ -0,0 +1,86 @@
+import { Button, CellStack, icons } from "@argent/ui"
+import { Center } from "@chakra-ui/react"
+import { FC } from "react"
+import { Link, To } from "react-router-dom"
+
+import { ShieldHeader } from "./ui/ShieldHeader"
+import { ChangeGuardian } from "./usePendingChangingGuardian"
+
+const { ArgentShieldIcon, ArgentShieldDeactivateIcon, TickIcon } = icons
+
+export interface ShieldBaseFinishScreenProps {
+  accountName?: string
+  pendingChangeGuardian?: ChangeGuardian
+  guardian?: string
+  returnRoute: To
+}
+
+export const ShieldBaseFinishScreen: FC<ShieldBaseFinishScreenProps> = ({
+  accountName,
+  pendingChangeGuardian,
+  guardian,
+  returnRoute,
+}) => {
+  const icon = pendingChangeGuardian
```

Comment:
I like ternary operator, but this feels a little complicated.
Maybe is more readable to just have an if condition?

```suggestion
let icon =  null
if (pendingChangeGuardian) {
   icon = ChangeGuardian.ADDING ? ArgentShieldIcon : ArgentShieldDeactivateIcon
}
else {
  icon = guardian ? TickIcon : ArgentShieldDeactivateIcon
}
```